client/src/partials/fiscal/fiscal.manage.js

```diff
@@ -0,0 +1,99 @@
+angular.module('bhima.controllers')
+  .controller('FiscalManagementController', FiscalManagementController);
+
+FiscalManagementController.$inject = [
+  '$state', 'FiscalService', 'NotifyService', 'util', 'moment'
+];
+
+/**
+ * This controller is responsible for creating and updating a fiscal year. It provides
+ * utility functions for submission and error handling.
+ */
+function FiscalManagementController($state, Fiscal, Notify, util, moment) {
+  var vm = this;
+
+  // identifier
+  var id = $state.params.id;
+
+  // global variables
+  vm.fiscal = {};
+  vm.submit = submit;
+  vm.maxLength = util.maxTextLength;
+  vm.numberOfMonths = numberOfMonths;
+
+  /**
+   * @function init
+   * @description init data for the module
+   */
+  function init() {
+
+    if (id && $state.current.name === 'fiscal.update') {
+      // concerned fiscal year
+      Fiscal.read(id)
+      .then(function (fiscalYear) {
+        vm.fiscal = fiscalYear;
+        vm.fiscal.start_date = new Date(vm.fiscal.start_date);
+        vm.fiscal.end_date = new Date(vm.fiscal.end_date);
+      })
+      .catch(Notify.handleError);
+    }
+
+    // previous fiscal year
+    Fiscal.read(null, { detailed: 1})
+    .then(function (previous) {
+      if (!previous.length) { return ; }
+
+      vm.previous_fiscal_year = previous.map(function (fy) {
+        fy.hrLabel = fy.label
+          .concat(' (', moment(fy.start_date).format('DD MMM YYYY').toString(), ' - ')
+          .concat(moment(fy.end_date).format('DD MMM YYYY').toString(), ')');
+        return fy;
+      });
+
+    })
+    .catch(Notify.handleError);
+  }
+
+  /**
+   * @function numberOfMonths
+   * @description get the number of months between two dates
+   */
+  function numberOfMonths() {
+    if (!vm.fiscal) { return ; }
+
+    var start_date = moment(vm.fiscal.start_date);
+    var end_date = moment(vm.fiscal.end_date);
+    vm.fiscal.number_of_months = Math.ceil(end_date.diff(start_date, 'months', true));
+  }
+
+  /**
+   * @method submit
+   * @description submit the form
+   */
+  function submit(form) {
+
+    // ensure all Angular form validation checks have passed
+    if (form.$invalid) {
+       Notify.danger('FORM.ERRORS.RECORD_ERROR');
+      return;
+    }
+
+    // get the number of months
+    numberOfMonths();
+
+    var isUpdate = (id && $state.current.name === 'fiscal.update');
```

Comment:
You could set this variable at the very top of the controller.  Like this:
```js
var id = $state.params.id;
var isUpdate = id && $state.current.name === 'fiscal.update'
```

Then you can use the `isUpdate` variable wherever you use the equality `(id && $state.current.name === 'fiscal.update')`.